drivers/serial/uart_rpi_pico.c

```diff
@@ -14,62 +15,313 @@
 
 struct uart_rpi_config {
 	uart_inst_t *const uart_dev;
-	uint32_t baudrate;
 	const struct pinctrl_dev_config *pcfg;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_config_func_t irq_config_func;
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+struct uart_rpi_data {
+	uint32_t baudrate;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_cb;
+	void *irq_cb_data;
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 };
 
+#define DEV_CFG(dev)  \
+	((const struct uart_rpi_config *const)(dev)->config)
+#define DEV_DATA(dev) \
+	((struct uart_rpi_data *const)(dev)->data)
+#define DEV_UART_INST(dev) \
+	((uart_inst_t *const)((const struct uart_rpi_config *const)(dev)->config)->uart_dev)
+#define DEV_UART_HW(dev) \
+	uart_get_hw(DEV_UART_INST(dev))
```

Comment:
please remove this sort of accessors, ref. https://github.com/zephyrproject-rtos/zephyr/pull/41918

Reply:
Ok, I removed all accessors.